sdk/src/main/scala/io/horizen/account/state/CertificateKeyRotationMsgProcessor.scala

```diff
@@ -143,6 +143,10 @@ case class CertificateKeyRotationMsgProcessor(params: NetworkParams) extends Nat
 
     if (!newKeySignature.isValid(keyRotationProof.newKey, newKeyAsMessage))
       throw new ExecutionRevertedException(s"Key rotation proof - self signature is invalid: $index")
+
+  } recoverWith {
+     case t : Exception =>
+     Failure(t)
```

Comment:
`recoverWith` is not needed anymore. Here we just throw the same exception.

Reply:
Right, done